frontend/apps/reader/modules/readerhyphenation.lua

```diff
@@ -26,17 +26,30 @@ function ReaderHyphenation:init()
                 text = v.name,
                 callback = function()
                     self.hyph_alg = v.filename
+                    self.ui.doc_settings:saveSetting("hyph_alg", self.hyph_alg)
                     UIManager:show(InfoMessage:new{
                         text = T(_("Changed hyphenation to %1."), v.name),
                     })
                     self.ui.document:setHyphDictionary(v.filename)
                     self.ui.toc:onUpdateToc()
                 end,
                 hold_callback = function()
-                    UIManager:show(ConfirmBox:new{
-                        text = T( _("Set fallback hyphenation to %1?"), v.name),
-                        ok_callback = function()
+                    UIManager:show(MultiConfirmBox:new{
+                        -- No real need for a way to remove default one, we can just
+                        -- toggle between setting a default OR a fallback (if a default
+                        -- one is set, no fallback will ever be used - if a fallback one
+                        -- is set, no default is wanted; so when we set one below, we
+                        -- remove the other).
```

Comment:
We might want to make this clear in the dialog somehow. The "original" says `Characters not found in the active font are shown in the fallback font instead.`

I'm thinking something along these lines: `"Set default or fallback hyphenation pattern to %1? Default will always take precedence while fallback will only be used if the language of the book can't be automatically determined"`